frontend/src/components/UserForm/UserForm.js

```diff
@@ -0,0 +1,198 @@
+import { useMutate } from "../../hooks/use-mutate"
+import { useRef, useState, useEffect} from "react"
+import { useUser } from "../../hooks/user-hooks"
+import { useParams } from "react-router"
+import { CountryDropdown, RegionDropdown } from "react-country-region-selector";
+
+const UserForm = () => {
+  const formRef = useRef()
+  const { updateUser } = useMutate()
+  const { id } = useParams()
+  const user = useUser({ "id": id })
+  const isAdmin = user?.data?.data?.data.user.admin
+
+  const fetchedCountry = user?.data?.data?.data?.user.country;
+  const fetchedState = user?.data?.data?.data?.user.state;
+  const [country, setCountry] = useState("");
+  const [region, setRegion] = useState("");
+
+  useEffect(() => {
+    if (fetchedCountry) setCountry(fetchedCountry);
+    if (fetchedState) setRegion(fetchedState);
+  }, [fetchedCountry, fetchedState]);
+
+  if (user.isError) return <h1>Something went wrong!</h1>
+  if (user.isLoading) return <h1>Loading...</h1>
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    // Add the current country/state values to the form data
+    formRef.current.append("country", country);
+    formRef.current.append("state", region);
+
+    const formData = new FormData(formRef.current)
+    const data = Object.fromEntries(formData)
+    const adminValue = data.admin === 'true' ? true : false;
+    data.country = country;
+    data.state = region;
+    if (!id) {
+        console.log('No user found')
+    } else {
```

Comment:
Is this necessary? no ID is implied as there will not be a creation function to this form.
Also console.log 😬

Reply:
you're right, not needed anymore, removed